src/renderer/src/hooks/useAuthentication.ts

```diff
@@ -0,0 +1,65 @@
+import { useCallback } from 'react'
+
+import { useTranslation } from 'react-i18next'
+
+import { ToastHelper } from '@renderer/helpers/ToastHelper'
+
+import { useCurrentLoginSessionSelector } from '@renderer/hooks/useAuthSelector'
+
+import { IAccountState } from '@shared/types/store'
+
+import { useModalNavigate } from './useModalRouter'
+
+type TUseAuthenticationParams = {
+  account: IAccountState
+}
+
+export const useAuthentication = () => {
+  const { modalNavigate, modalErase } = useModalNavigate()
+  const { t } = useTranslation('hooks', { keyPrefix: 'useAuthentication' })
+  const { currentLoginSessionRef } = useCurrentLoginSessionSelector()
+
+  const authenticate = useCallback(
+    async ({ account }: TUseAuthenticationParams) => {
+      if (
+        account.type === 'standard' &&
+        currentLoginSessionRef.current?.type !== 'hardware' &&
+        currentLoginSessionRef.current?.encryptedPassword
+      ) {
+        const session = currentLoginSessionRef.current
+
+        return new Promise<void>((resolve, reject) => {
+          modalNavigate('confirm-password-transaction', {
+            replace: true,
+            state: {
+              account,
+              onSubmitPassword: async (password: string) => {
+                const encryptedPassword = session.encryptedPassword
+                const decryptedPassword = await window.api.sendAsync('decryptBasedOS', encryptedPassword)
+
+                if (!password || password !== decryptedPassword) {
+                  throw new Error(t('errors.invalidPassword'))
+                }
+                resolve()
+                modalErase('side')
```

Comment:
You are already doing this validation within the modal, so it is not necessary here